### PR TITLE
Ensure that we show all applicable refactorings for lightbulb under t…

### LINKED ISSUE #44594: Rename object/variable names no longer available on the context menu
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1043991/rename-objectvariable-names-no-longer-available-on.html)._

---
[regression] [worked-in:All versions prior to Version 16.6.0]
Renaming an object or variable name in Visual Studio Version 16.6.0 no longer provides the 'rename' option on the context menu. It's no longer possible to effectively and efficiently rename objects as a bulk operation.

---
#### David Bentley on 5/21/2020, 08:16 PM: 

<p>Just updated to 16.6.0 and having the same issue. Was driving me nuts thinking it was just me. As a side note, [Ctrl + R], [Ctrl + R] allows to rename, but you have to use it BEFORE you rename and not after. It is just not as efficient or familiar.</p>

#### Visual Studio Feedback System on 5/22/2020, 01:53 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### dave on 5/22/2020, 06:06 AM: 

<p>Yes. I don't know if that [CTRL+R]^2 functionality has always been there, but it's un-intuitive and not nearly as efficient. Especially with the extra step of needing to click 'Apply' on a dialog that is on the other side of the screen. <br>There was nothing wrong with the functionality in prior versions, rename the object, right+click and select Rename. It worked, it was effective, it wasn't broken. <br>That needs to be put back. Right now that missing functionality is simply a bug.</p>

#### David Bentley on 5/22/2020, 06:56 AM: 

<p>Looks like this was around in 16.5.0 and was claimed to be fixed (but clearly has not):<br><a target='_blank' href="https://github.com/dotnet/roslyn/issues/42703">https://github.com/dotnet/roslyn/issues/42703</a></p>

#### Visual Studio Feedback System on 5/27/2020, 09:55 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by mavasani**
…he caret

Fixes #44594 

Verified that rename tracking code action now shows up in the lightbulb under the caret

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 1]
```diff
@@ -799,7 +818,11 @@ private static SuggestedActionSetPriority GetSuggestedActionSetPriority(CodeActi
                             document, selection, isBlocking: true, addOperationScope, cancellationToken),
                         cancellationToken).WaitAndGetResult(cancellationToken);
 
-                    var filteredRefactorings = FilterOnUIThread(refactorings, workspace);
+                    // If we are computing refactorings outside the 'Refactoring' context, i.e. for example, from the lightbulb under a squiggle or selection,
+                    // then we want to filter out refactorings outside the selection span.
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:822]: i would explain 'why' (or link to an issue).

File: src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.StateMachine.cs [hunk 2]
```diff
@@ -308,12 +308,13 @@ internal int StoreCurrentTrackingSessionAndGenerateId()
                                 trackingSession.OriginalName,
                                 snapshotSpan.GetText());
 
-                            return new RenameTrackingCodeAction(
-                                document, title, refactorNotifyServices, undoHistoryRegistry);
+                            return (new RenameTrackingCodeAction(
+                                        document, title, refactorNotifyServices, undoHistoryRegistry),
+                                    snapshotSpan.Span.ToTextSpan());
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.StateMachine.cs:313]:
```suggestion
                            return (new RenameTrackingCodeAction(document, title, refactorNotifyServices, undoHistoryRegistry),
                                    snapshotSpan.Span.ToTextSpan());
```

**Comment by CyrusNajmabadi** [comment]
Would *really* like an integration test here.

File: src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.StateMachine.cs (hunk 2 above)
Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.StateMachine.cs:313]: Added a note in https://github.com/dotnet/roslyn/issues/44855

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs (hunk 1 above)
Comment (mavasani) [reply src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:822]: Added a note in https://github.com/dotnet/roslyn/issues/44855

**Comment by CyrusNajmabadi** [comment]
> Hope that is fine.

Please do prioritize though. It is important we don't regress this. 

I assume you did manually validate though?

**Comment by mavasani** [comment]
> I assume you did manually validate though?

Yes, I did!

> Please do prioritize though. It is important we don't regress this

Yep, investigating now.

**Pull request merged**